lib/ansible/modules/packaging/os/swupd.py

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/python
+
+# (c) 2017, Alberto Murillo <alberto.murillo.silva@intel.com>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: swupd
+short_description: Manages bundles with M(swupd).
+description:
+  - Manages bundles with the M(swupd) update manager, which is used by the
+    Clear Linux Project for Intel Architecture.
+version_added: "2.3"
+author: Alberto Murillo (@albertomurillo)
+options:
+  contenturl:
+    description:
+      - URL for content file download.
+    required: false
+    default: null
+  format:
+    description:
+      - The format suffix for version file downloads.
+    required: false
+    default: null
+  manifest:
+    description:
+      - Manifest to verify against to.
+    required: false
+    default: null
+    aliases: [release, version]
+  name:
+    description:
+      - Name of the (I)bundle to install or remove.
+    required: false
+    default: null
+    aliases: [bundle]
+  state:
+    description:
+      - Indicates the desired (I)bundle state. C(present) ensures the bundle
+        is installed while C(absent) ensures the (I)bundle is not installed.
+    required: false
+    default: present
+    choices: [present, absent]
+  update:
+    description:
+      - Updates the OS to the latest version.
+    required: false
+    default: no
+  url:
+    description:
+      - Overrides both I(contenturl) and I(versionurl).
+    required: false
+    default: null
+  verify:
+    description:
+      - Verify content for OS version.
+    required: false
+    default: null
+  versionurl:
+    description:
+      - URL for version string download.
+    required: false
+    default: null
+'''
+
+EXAMPLES = '''
+- name Update the OS to the latest version
+  swupd:
+    update: yes
+
+- name: Installs the "foo" bundle
+  swupd:
+    name: foo
+    state: present
+
+- name: Removes the "foo" bundle
+  swupd:
+    name: foo
+    state: absent
+
+- name: Check integrity of filesystem
+  swupd:
+    verify: yes
```

Comment:
Does this return anything, such as where there are validation errors?
An of how to access that could be useful (see `register:` and `debug`)